src/main/java/seedu/address/model/money/Money.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.model.money;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's Money Balance in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidMoney(String)}
+ */
+public class Money {
+    public static final String MESSAGE_MONEY_CONSTRAINTS = "Money values should be numbers";
+    public static final String MONEY_VALIDATION_REGEX = "-?\\d+(\\.\\d+)?";
+
+    public final Double balance;
```

Comment:
I think you use a Class Double here, instead of `double` variable type. just be aware of this.